src/ripple/basics/base_uint.h

```diff
@@ -64,6 +65,13 @@ struct is_contiguous_container<Slice> : std::true_type
 
 }  // namespace detail
 
+template <std::size_t Bits, typename Tag>
+class base_uint;
+
+template <std::size_t Bits, typename Tag>
+[[nodiscard]] constexpr bool
+operator==(base_uint<Bits, Tag> const& lhs, base_uint<Bits, Tag> const& rhs);
+
```

Comment:
My intention was to be able to compare the arrays `data_` with `std::uint32_t` data elements instead of using `std::mismatch()` with `unsigned char` data elements. It's probably an overkill. I revert this change and take your suggested code.

Reply:
Ah, I see what you were trying to do. It makes sense, but you need to be careful (and you were). This trick will _ONLY_ work for equality/inequality comparisons, because the `data_` array is stored in _big endian_ order, which on platforms like the x86 is **NOT** the native endian order. Relational comparisons (gt/gte/lt/lte) break. Consider, for example, the following two 32-bit unsigned values:

    a = 256, b = 1

If you convert them into big endian they look like 

    x = 65536, y = 16777216

Notice what happened... `a > b` but `x < y`. Oops!

I think that having the "bespoke" `operator==` is fine although I doubt there are any performance gains to be had. But you should definitely consider adding a comment explaining that this is safe, despite the fact that the individual limbs (the entries in the `data_` array) are in big endian, because we are comparing for _equality_/_inequality_.

The existing `compare` function **CANNOT** do that trick and _MUST_ use `std::mismatch`. It could also use a comment too, explaining that this works because the pointer returned from `cbegin` is a byte-based view of the `data_` which is actually in big-endian order.

Reply:
You are right. This is quite tricky. I use `std::mismatch` for now.